ipatests/test_integration/test_replica_promotion.py

```diff
@@ -527,3 +527,58 @@ def test_replica_install_with_existing_entry(self):
         master.run_command(arg)
 
         tasks.install_replica(master, replica)
+
+
+class TestNsslapdIgnoreTimeSkew(IntegrationTest):
+    """value for nsslapd-ignore-time-skew should remain off by default for
+    dirsrv. However while installing replica, it shuold be on. This test
+    checks that while installing replica, value for nsslapd-ignore-time-skew
+    set to on and after set to off.
+
+    https://pagure.io/freeipa/issue/7211"""
+
+    num_replicas = 2
+
+    def test_nsslapd_ignore_time_skew(self):
+        master = self.master
+        tasks.install_master(master)
+        replica = self.replicas[0]
+        client = self.replicas[1]
+
+        tasks.install_replica(master, replica)
+        log_file = '/var/log/ipareplica-install.log'
+        replica_install_log = replica.get_file_contents(log_file)
+        regex = '^.*nsslapd\-ignore\-time\-skew:\n(.*)$'
+        occ_list = re.findall(regex, replica_install_log, re.M)
+        assert 'on' in occ_list[0] and 'off' in occ_list[1]
+
+        # check on replica promotion
+        tasks.install_client(master, client)
+        args = ['ipa-replica-install', '-U',
+                '-p', master.config.dirman_password,
+                '-w', master.config.admin_password,
+                '--ip-address', client.ip]
+        result = client.run_command(args)
+        assert result.returncode == 0
+
+        replica_install_log = client.get_file_contents(log_file)
+        occ_list = re.findall(regex, replica_install_log, re.M)
+        assert 'on' in occ_list[0] and 'off' in occ_list[1]
+
+        # check if nsslapd-ignore-time-skew is set to off for disrv
+        arg = ['ldapsearch', '-D',
+               str(replica.config.dirman_dn),   # pylint: disable=no-member
```

Comment:
Why the `str` all of a sudden?

Reply:
when ran without str:
[ipa.ipatests.pytest_plugins.integration.host.Host.replica.OpenSSHTransport] RUN ['ldapsearch', '-D', ipapython.dn.DN('cn=Directory Manager'), '-w', 'Secret123', '-h', 'replica.testrelm.test', '-b', 'cn=config']
[ipa.ipatests.pytest_plugins.integration.host.Host.replica.cmd54] RUN ['ldapsearch', '-D', ipapython.dn.DN('cn=Directory Manager'), '-w', 'Secret123', '-h', 'replica.testrelm.test', '-b', 'cn=config']
FAILED[ipa.ipatests.pytest_plugins.integration.host.Host.master.OpenSSHTransport] RUN ['kinit', 'admin']